agent/lib/kontena/workers/log_worker.rb

```diff
@@ -20,14 +20,15 @@ class LogWorker
     QUEUE_MAX_SIZE = 2000
     QUEUE_THROTTLE = (QUEUE_MAX_SIZE * 0.8)
     WATCH_INTERVAL = 10.0
+    BATCH_SIZE = 10
 
     def websocket_client
       Celluloid::Actor[:websocket_client]
     end
 
-    # @param [Boolean] autostart
+    # @param autostart [Boolean]
     def initialize(autostart = true)
-      @queue = Queue.new
+      @queue = []
```

Comment:
Feels a bit risky but ok with the [MRI GIL](https://www.jstorimer.com/blogs/workingwithcode/8100871-nobody-understands-the-gil-part-2-implementation). I don't know how many other places in the agent are dependent on the GIL, in terms of preventing us from switching to alternative runtimes?

Reply:
No plans for alternative runtimes.

Reply:
Btw concurrent-ruby has [Array](http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/Array.html).